Bootstrap compiler on beta needs an update

Naively updating the bootstrap compiler yields:

```
error[E0658]: mod statements in non-mod.rs files are unstable (see issue #44660)
 --> /home/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/chalk-engine-0.8.0/src/context.rs:7:16              
  |                                                                                                                 
7 | pub(crate) mod prelude;                                                                                         
  |                ^^^^^^^                                                                                          
  |                                                                                                                 
  = help: add #![feature(non_modrs_mods)] to the crate attributes to enable                                         
  = help: on stable builds, rename this file to context/mod.rs                                                      
```

I believe this was due to the last minute destabilization of this feature for the 1.30.0 release.

@nikomatsakis could a point release of the `chalk-engine` crate be made which builds on stable?

--- a/src/stage0.txt
+++ b/src/stage0.txt
@@ -12,9 +12,9 @@
 # source tarball for a stable release you'll likely see `1.x.0` for rustc and
 # `0.x.0` for Cargo where they were released on `date`.
 
-date: 2018-10-13
-rustc: beta
-cargo: beta
+date: 2018-10-25
+rustc: 1.30.0
+cargo: 0.31.0
 
 # When making a stable release the process currently looks like:
 #

